src/files-card-body.tsx

```diff
@@ -363,9 +371,28 @@ export const FilesCardBody = ({
           ref={folderViewRef}
         >
             {contextMenu}
-            {sortedFiles.length === 0 &&
+            {sortedFiles.length === 0 && currentFilter &&
                 <EmptyStatePanel
-                  paragraph={currentFilter ? _("No matching results") : _("Directory is empty")}
+                  icon={SearchIcon}
+                  title={_("No matching results")}
+                  action={_("Clear filter")}
+                  onAction={() => setCurrentFilter("")}
+                  actionVariant="link"
+                />}
+            {sortedFiles.length === 0 && !currentFilter &&
+                <EmptyStatePanel
+                  icon={FolderIcon}
+                  title={_("Directory is empty")}
+                  paragraph={hiddenFilesCount === 0
+                      ? null
+                      : cockpit.format(
+                          cockpit.ngettext(
+                              "$0 item is hidden", "$0 items are hidden", hiddenFilesCount
+                          ), hiddenFilesCount
+                      )}
+                  action={_("Show hidden items")}
+                  onAction={() => setShowHidden(true)}
```

Comment:
It seems to me that this should also be conditionalized on `hiddenFilesCount` for consistency?

Reply:
The action? Uhmmm yes woops!